File: src/test/java/ac/dnd/server/admission/infrastructure/persistence/entity/ApplicantTest.java
package ac.dnd.server.admission.infrastructure.persistence.entity;

import static org.assertj.core.api.AssertionsForClassTypes.*;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import ac.dnd.server.annotation.UnitTest;
import ac.dnd.server.enums.ApplicantStatus;
import ac.dnd.server.enums.ApplicantType;
import ac.dnd.server.fixture.ApplicantFixture;
import ac.dnd.server.fixture.EventFixture;

@UnitTest
class ApplicantTest {

	@Test
	void Applicant_객체를_생성하고_필드_값을_확인한다() {
		final String name = "홍길동";
		final String email = "gildong@example.com";
		final ApplicantType type = ApplicantType.FRONTEND;
		final ApplicantStatus status = ApplicantStatus.PASSED;

		final Applicant applicant = new Applicant(
			name,
			email,
			name,
			email,
			type,
			status
		);

		assertThat(applicant.getName()).isEqualTo(name);
		assertThat(applicant.getEmail()).isEqualTo(email);
		assertThat(applicant.getType()).isEqualTo(type);
		assertThat(applicant.getStatus()).isEqualTo(status);
		assertThat(applicant.getEvent()).isNull(); // 아직 Event가 설정되지 않음
	}

	@Test
	void withEvent_메서드로_Applicant에_Event를_설정한다() {
		final Applicant applicant = ApplicantFixture.create();
		final Event event = EventFixture.create();

		applicant.withEvent(event);

		Assertions.assertNotNull(applicant.getEvent());
		assertThat(applicant.getEvent()).isEqualTo(event);
	}
}

File: src/test/kotlin/ac/dnd/server/admission/infrastructure/persistence/entity/ApplicantTest.kt
package ac.dnd.server.admission.infrastructure.persistence.entity

import ac.dnd.server.annotation.UnitTest
import ac.dnd.server.enums.ApplicantStatus
import ac.dnd.server.enums.ApplicantType
import ac.dnd.server.fixture.ApplicantFixture
import ac.dnd.server.fixture.EventFixture
import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe
import io.kotest.matchers.shouldNotBe

@UnitTest
class ApplicantTest : DescribeSpec({

    describe("Applicant") {

        it("Applicant 객체를 생성하고 필드 값을 확인한다") {
            // given
            val name = "홍길동"
            val email = "gildong@example.com"
            val type = ApplicantType.FRONTEND
            val status = ApplicantStatus.PASSED

            // when
            val applicant = Applicant(
                name,
                email,
                name,
                email,
                type,
                status
            )

            // then
            applicant.name shouldBe name
            applicant.email shouldBe email
            applicant.type shouldBe type
            applicant.status shouldBe status
            applicant.event shouldBe null
        }

        it("withEvent 메서드로 Applicant에 Event를 설정한다") {
            // given
            val applicant = ApplicantFixture.create()
            val event = EventFixture.create()

            // when
            applicant.withEvent(event)

            // then
            applicant.event shouldNotBe null
            applicant.event shouldBe event
        }
    }
})